src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -111,26 +111,46 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
     /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
     /// </remarks>
     public HttpRequestInterceptionBuilder For(params Predicate<HttpRequestMessage>[] predicates)
+    {
+        return For(predicates
+            .Select(predicate => new Func<HttpRequestMessage, Task<bool>>((message) => Task.FromResult(predicate(message))))
+            .ToArray());
+    }
+
+    /// <summary>
+    /// Configures the builder to match any request that meets all the criteria defined by the specified predicates.
+    /// </summary>
+    /// <param name="predicates">
+    /// Two or more delegates to a method which returns <see langword="true"/> if the
+    /// request is considered a match; otherwise <see langword="false"/>.
+    /// </param>
+    /// <returns>
+    /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+    /// </returns>
+    /// <remarks>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </remarks>
+    public HttpRequestInterceptionBuilder For(params Func<HttpRequestMessage, Task<bool>>[] predicates)
     {
         if (predicates?.Length == 1)
         {
             return For(predicates[0]);
         }
 
-        _requestMatcher = predicates != null ?
-        (p) =>
-        {
-            foreach (var predicate in predicates)
+        _requestMatcher = predicates == null || predicates.Length == 0
```

Comment:
I think `Length == 0` should be an exception because it's incorrect usage. `For()` wouldn't do anything and be a mistake, so shouldn't really clear the existing registrations either.